### PR TITLE
In list delete update

### PR DESCRIPTION
**Pull request by ChianHuei**
1. The EDIT buttons on listCard are functional.
User can upload cover images for lists and change list title, description and cover image.
Screen Shot 2020-07-20 at 9 45 21 PM

2. added REMOVE button on listCard
When clicking remove button, a dialog pops up asking user for confirmation.
Screen Shot 2020-07-20 at 9 45 52 PM

### CODE REVIEW

File: client/src/components/ProductCard.js [hunk 1]
```diff
@@ -59,31 +58,32 @@ export default function ProductCard(props) {
...
+                var errmess = new Error(error.message);
+                throw errmess;
         })
         .then(response => response.json())
-        .then(res=>{shListsContext.hideProduct(props.product);
+        .then(res=>{
+            shListsContext.hideProduct(props.product);
             shListsContext.handleProductDeletion(props.listId,props.product._id);})
-        .catch(error =>console.log('product-deleting failed', error.message));
+        .catch(error =>alert('product-deletion failed', error.message));
```
Comment (surhud004) [inline client/src/components/ProductCard.js:76]: Alert is fine for now. I think we should use a better error displaying method. There is a good library which I have used before - `connect flash` - https://www.npmjs.com/package/connect-flash

File: client/src/components/ShoppingLists.js [hunk 2]
```diff
@@ -46,7 +46,7 @@ const ShoppingLists = ()=>{
                 </Typography>
                 <br/>
                 <Grid container spacing={5}>
-                {shListsContext.lists.map((list) => (
+                {shListsContext.lists.slice().reverse().map((list) => (
```
Comment (surhud004) [inline client/src/components/ShoppingLists.js:49]: We can use this for now. Actually we have created date as a parameter in List schema. We can sort descending through Mongoose and pass data to FE alternatively.

File: client/src/components/ProductCard.js (hunk 1 above)
Comment (ChianHuei) [reply client/src/components/ProductCard.js:76]: I agree. I don't like `alert`.

**Review by surhud004 — APPROVED** [review]
Looks good otherwise!!

File: client/src/components/ShoppingLists.js (hunk 2 above)
Comment (ChianHuei) [reply client/src/components/ShoppingLists.js:49]: Great! I will try.

**Comment by Ethorf** [comment]
Looks great to me!

**Pull request merged**